### PR TITLE
add _.of and _.fromError

### PR DESCRIPTION
**Pull request by svozza**
As discussed in #519.

### CODE REVIEW

File: lib/index.js [hunk 1]
```diff
@@ -771,6 +771,41 @@ function Stream(/*optional*/xs, /*optional*/secondArg, /*optional*/mappingHint)
...
+ *
+ * _.fromError(new Error('Single Error')).toCallback(function (err, result) {
+ *     // err contains Error('Single Error') object
+ * }
+ */
+
+_.fromError = function (error) {
+    return _(function (push) {
+        push(error);
+        push(_.nil);
```
Comment (vqvu) [inline lib/index.js]: `push(null, _.nil)`

Comment (svozza) [inline lib/index.js]: Doh!

**Comment by vqvu** [comment]
LGTM.

**Comment by vqvu** [comment]
Can you add a short entry to the `CHANGELOG.md` referencing this change?

**Comment by svozza** [comment]
Oh yes, of course.

**Comment by svozza** [comment]
Done.

**Comment by svozza** [comment]
I've just realised, have I made this function too dumb, i.e., should I check to see if what I've got is a Promise or an Iterable and not wrap those objects in `[]` before passing them to the Highland constructor?

**Comment by vqvu** [comment]
I don't think so, for three reasons

1. Arrays are Iterables, and it seems weird to have to say `_.of([[1, 2,
3]])` to get a stream of arrays. It's not very intuitive.
2. You could argue for promises, but that functionality may be better off
implemented as part of `asyncify` instead. People already have `_` if they
know for sure that they have a promise.
3. Special cases would permanently break compatibility with fantasy land's
definition of `of`.

**Comment by svozza** [comment]
Yep, makes sense. Looking at the Fantasy Land spec though it looks like `of` needs to be on the object's prototype so we'll have to revisit this later anyway:

https://github.com/fantasyland/fantasy-land#of-method

**Pull request merged**